tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py

```diff
@@ -21,42 +23,44 @@
 
 @pytest.mark.parametrize(
     "dtype",
-    (ttl.tensor.DataType.BFLOAT16, ttl.tensor.DataType.FLOAT32),
+    (ttnn.bfloat16, ttnn.float32),
     ids=["bfloat16", "float"],
 )
 @pytest.mark.parametrize("inplace", [True, False])
 def test_softmax(device, inplace, dtype):
-    if is_grayskull() and dtype == ttl.tensor.DataType.FLOAT32:
+    if is_grayskull() and dtype == ttnn.float32:
         pytest.skip("Skipping float32 tests on Grayskull")
 
     torch.manual_seed(0)
-    sm_op = ttl.operations.primary.softmax_in_place if inplace else ttl.tensor.softmax
+    sm_op = ttnn.softmax_in_place if inplace else ttnn.softmax
```

Comment:
why do we have softmax_in_place and softmax?

Doesn't software support optional output tensor? In that case inplace is just passing input there

Reply:
that's some quite old code, which I don't know why they are separate APIs. 
What does optional output tensor do? does it allow tensors to be placed on the same L1 address?

Reply:
If you already have an allocated tensor which you can reuse, you can pass it and operation will put result into it. So `optional output tensor` allows to eliminate tensor allocation.